src/test/java/com/hazelcast/kubernetes/HazelcastKubernetesDiscoveryStrategyFactoryTest.java

```diff
@@ -53,6 +53,9 @@
     public void setup()
             throws Exception {
         PowerMockito.whenNew(KubernetesClient.class).withAnyArguments().thenReturn(client);
+        PowerMockito.spy(KubernetesConfig.class);
```

Comment:
How about setting a KubernetesProperty `NAMESPACE` instead of using PowerMockito here? 

And the same for the other test.

Reply:
Created a specific test where we use the mounted K8s files and avoided reading the file if the namespace property is set.